packages/mdl-ripple/util.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export function supportsCssVariables(windowObj) {
+  // NOTE: This does not work in Safari *on purpose* (see: https://bugs.webkit.org/show_bug.cgi?id=154669)
+  // See the README for more info.
+  return windowObj.CSS && typeof windowObj.CSS.supports === 'function' &&
+      windowObj.CSS.supports('--css-vars', 'yes');
```

Comment:
If you want to feature-detect everything with proper support for custom variables plus Safari 10 (not 9), you can do:
```js
CSS.supports('--foo', 'red') || (CSS.supports('(--foo: red)') && CSS.supports('color', '#00000000'));
```

This makes use of the fact that Safari 10 supports `#RRGGBBAA` color values, whereas Safari 9 doesn't.

Doing this would at least give us ripples in Safari 10, where they should work properly.